go-controller/pkg/ovn/loadbalancer/loadbalancer.go

```diff
@@ -0,0 +1,194 @@
+			"external_ids:k8s-cluster-lb-tcp=yes")
+	} else if protocol == kapi.ProtocolUDP {
+		out, _, err = util.RunOVNNbctl("--data=bare", "--no-heading",
+			"--columns=_uuid", "find", "load_balancer",
+			"external_ids:k8s-cluster-lb-udp=yes")
+	} else if protocol == kapi.ProtocolSCTP {
+		out, _, err = util.RunOVNNbctl("--data=bare", "--no-heading",
+			"--columns=_uuid", "find", "load_balancer",
+			"external_ids:k8s-cluster-lb-sctp=yes")
+	}
+	if err != nil {
+		return "", err
+	}
+	if out == "" {
+		return "", fmt.Errorf("no load balancer found in the database")
+	}
+	return out, nil
+}
+
+// GetLoadBalancerVIPs returns a map whose keys are VIPs (IP:port) on loadBalancer
+func GetLoadBalancerVIPs(loadBalancer string) (map[string]interface{}, error) {
+	outStr, _, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"get", "load_balancer", loadBalancer, "vips")
+	if err != nil {
+		return nil, err
+	}
+	if outStr == "" {
+		return nil, fmt.Errorf("load balancer vips in OVN DB is an empty string")
+	}
+	// sample outStr:
+	// - {"192.168.0.1:80"="10.1.1.1:80,10.2.2.2:80"}
+	// - {"[fd01::]:80"="[fd02::]:80,[fd03::]:80"}
+	outStrMap := strings.Replace(outStr, "=", ":", -1)
+
+	var raw map[string]interface{}
+	err = json.Unmarshal([]byte(outStrMap), &raw)
+	if err != nil {
+		return nil, err
+	}
+	return raw, nil
+}
+
+// DeleteLoadBalancerVIP removes the VIP as well as any reject ACLs associated to the LB
+func DeleteLoadBalancerVIP(loadBalancer, vip string) error {
+	vipQuotes := fmt.Sprintf("\"%s\"", vip)
+	stdout, stderr, err := util.RunOVNNbctl("--if-exists", "remove", "load_balancer", loadBalancer, "vips", vipQuotes)
+	if err != nil {
+		// if we hit an error and fail to remove load balancer, we skip removing the rejectACL
+		return fmt.Errorf("error in deleting load balancer vip %s for %s"+
+			"stdout: %q, stderr: %q, error: %v",
+			vip, loadBalancer, stdout, stderr, err)
+	}
+	return nil
+}
+
+// UpdateLoadBalancer updates the VIP for sourceIP:sourcePort to point to targets (an
+// array of IP:port strings)
+func UpdateLoadBalancer(lb, sourceIP string, sourcePort int32, targets []string) error {
+	vip := util.JoinHostPortInt32(sourceIP, sourcePort)
+	lbTarget := fmt.Sprintf(`vips:"%s"="%s"`, vip, strings.Join(targets, ","))
+
+	out, stderr, err := util.RunOVNNbctl("set", "load_balancer", lb, lbTarget)
+	if err != nil {
+		return fmt.Errorf("error in configuring load balancer: %s "+
+			"stdout: %q, stderr: %q, error: %v", lb, out, stderr, err)
+	}
+
+	return nil
+}
+
+// GetLogicalRoutersForLoadBalancer get the switches associated to a LoadBalancer
+func GetLogicalSwitchesForLoadBalancer(lb string) ([]string, error) {
+	out, _, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=_uuid", "find",
+		"logical_switch", fmt.Sprintf("load_balancer{>=}%s", lb))
+	if err != nil {
+		return nil, err
+	}
+	if len(strings.Fields(out)) > 0 {
+		return strings.Fields(out), nil
```

Comment:
yeah, actually I think that we can avoid to check the error too and return `return strings.Fields(out), err` but I will keep it because it is easier to read.